### ISSUE TITLE
Issues with passing geometries to aggregate_spatial, CSV export

### ISSUE THREAD

**Issue by jonathom** [Open-EO/openeo-geopyspark-driver#76]
When I aggregate spatially with a MultiPolygon (via _python_ client), only one set of values is returned in CSV export. I'm not sure where this happens (aggregation or export), but I would expect all geometries in the exported file.

Example:
`{
  "process_graph": {
    "loadcollection1": {
      "arguments": {
        "bands": [
          "B08"
        ],
        "id": "SENTINEL2_L2A_SENTINELHUB",
        "spatial_extent": {
          "west": 5.590152740478516,
          "south": 52.66951218928503,
          "east": 5.753231048583984,
          "north": 52.75686657942532
        },
        "temporal_extent": [
          "2021-04-26",
          "2021-04-30"
        ]
      },
      "process_id": "load_collection"
    },
    "aggregatespatial1": {
      "arguments": {
        "data": {
          "from_node": "loadcollection1"
        },
        "geometries": {
          "coordinates": [
            [
              [
                [
                  5.645427,
                  52.702368
                ],
                [
                  5.6568,
                  52.702446
                ],
                [
                  5.645728,
                  52.716356
                ],
                [
                  5.645427,
                  52.702368
                ]
              ]
            ],
            [
              [
                [
                  5.67,
                  52.73
                ],
                [
                  5.682,
                  52.699
                ],
                [
                  5.645,
                  52.693
                ],
                [
                  5.67,
                  52.73
                ]
              ]
            ]
          ],
          "type": "MultiPolygon"
        },
        "reducer": {
          "process_graph": {
            "mean1": {
              "arguments": {
                "data": {
                  "from_parameter": "data"
                }
              },
              "process_id": "mean",
              "result": true
            }
          }
        }
      },
      "process_id": "aggregate_spatial"
    },
    "saveresult1": {
      "arguments": {
        "data": {
          "from_node": "aggregatespatial1"
        },
        "format": "CSV",
        "options": {}
      },
      "process_id": "save_result",
      "result": true
    }
  }
}`

Additionally, when I try to aggregate with a feature collection (directly from geojson.io, via _javascript_ client @m-mohr), I get the error `list index out of range`.

Example:
`{
  "process_graph": {
    "loadco1": {
      "arguments": {
        "bands": [
          "B08"
        ],
        "id": "SENTINEL2_L2A_SENTINELHUB",
        "spatial_extent": {
          "west": 4.320287704467773,
          "south": 50.8050668687832,
          "east": 4.392900466918944,
          "north": 50.84215341242185
        },
        "temporal_extent": [
          "2020-06-01",
          "2020-09-01"
        ]
      },
      "process_id": "load_collection"
    },
    "aggreg1": {
      "arguments": {
        "data": {
          "from_node": "loadco1"
        },
        "geometries": {
          "features": [
            {
              "geometry": {
                "coordinates": [
                  [
                    [
                      4.363503456115723,
                      50.83242375662908
                    ],
                    [
                      4.3399858474731445,
                      50.82705667183455
                    ],
                    [
                      4.369468688964844,
                      50.81493780212673
                    ],
                    [
                      4.363503456115723,
                      50.83242375662908
                    ]
                  ]
                ],
                "type": "Polygon"
              },
              "properties": {},
              "type": "Feature"
            }
          ],
          "type": "FeatureCollection"
        },
        "reducer": {
          "process_graph": {
            "mean1": {
              "arguments": {
                "data": {
                  "from_parameter": "data"
                }
              },
              "process_id": "mean",
              "result": true
            }
          }
        }
      },
      "process_id": "aggregate_spatial"
    },
    "savere1": {
      "arguments": {
        "data": {
          "from_node": "aggreg1"
        },
        "format": "CSV"
      },
      "process_id": "save_result",
      "result": true
    }
  }
}`

Just overall I don't understand how exactly geometries need to be passed here.

**Comment by m-mohr** [comment]
@jonathom 

The issue with the polygon/multipolygon we have discussed today and lead to the following PR: https://github.com/Open-EO/openeo-processes/pull/253

For the list index out of range issue I'm wondering whether you also get the issue if you use the same spatial and temporal extents and geometries as in the first process graph? Maybe that's the issue, because everything else looks very similar.

**Comment by jonathom** [comment]
> Maybe that's the issue, because everything else looks very similar.

No, still the same. See this [gist](https://gist.github.com/jonathom/c3b33c582bf71c958136e9846503c808). Passing a FeatureCollection still results into `list index out of range`.

What does work is passing this: 
`{
        "type": "Polygon",
        "coordinates": [
            [
            [
                5.645427703857422,
                52.70236859806736
            ],
            [
                5.656800270080566,
                52.70244661236617
            ],
            [
                5.645728111267089,
                52.71635693250797
            ],
            [
                5.645427703857422,
                52.70236859806736
            ]
            ]
        ]
        }`
But how could I pass multiple polygons without using a collection?

I also asked in [openeo-processes#253](https://github.com/Open-EO/openeo-processes/pull/253), how does one pass multiple geometries via python and JS?

**Comment by soxofaan** [comment]
I had a quick look and it turns out that the `list index out of range` error comes from the CSV export logic. If you download as JSON,  you van get the data just fine.

for reference, this is the stack trace I found in the logs:
```
  File "/data1/hadoop/yarn/local/usercache/openeo/appcache/application_1617264705177_33713/container_e4938_1617264705177_33713_01_000001/venv/lib64/python3.6/site-packages/openeo_driver/views.py", line 459, in result
    return execute()
  File "/data1/hadoop/yarn/local/usercache/openeo/appcache/application_1617264705177_33713/container_e4938_1617264705177_33713_01_000001/venv/lib64/python3.6/site-packages/openeo_driver/views.py", line 490, in execute
    return result.create_flask_response()
  File "/data1/hadoop/yarn/local/usercache/openeo/appcache/application_1617264705177_33713/container_e4938_1617264705177_33713_01_000001/venv/lib64/python3.6/site-packages/openeo_driver/save_result.py", line 172, in create_flask_response
    filename = self.to_csv()
  File "/data1/hadoop/yarn/local/usercache/openeo/appcache/application_1617264705177_33713/container_e4938_1617264705177_33713_01_000001/venv/lib64/python3.6/site-packages/openeo_driver/save_result.py", line 244, in to_csv
    date_band_dict[date] = list(map(lambda p: p[0], polygon_results))
  File "/data1/hadoop/yarn/local/usercache/openeo/appcache/application_1617264705177_33713/container_e4938_1617264705177_33713_01_000001/venv/lib64/python3.6/site-packages/openeo_driver/save_result.py", line 244, in <lambda>
    date_band_dict[date] = list(map(lambda p: p[0], polygon_results))
IndexError: list index out of range
```
